app/src/main/java/com/example/user/my_calculator/MainActivity.java

```diff
@@ -12,290 +12,132 @@
-                } else if(ACTION != null && ACTION.equals("*")){
-                    double result = val1 * Double.parseDouble(edtInput.getText().toString());
-                    txtResult.setText(null);
-                    edtInput.setText(String.valueOf(result));
-                } else if(ACTION != null && ACTION.equals("/")){
-                    double result = val1 / Double.parseDouble(edtInput.getText().toString());
-                    txtResult.setText(null);
-                    edtInput.setText(String.valueOf(result));
-                }
-
-                ACTION = null;
-                val1 = NaN;
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
-
-        btnNumber0.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "0");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
-
-        btnNumber1.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "1");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    public void onCEClick(View view) {
+        edtInput.setText(null);
+    }
 
-        btnNumber2.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "2");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    public void onCClick(View view) {
+        firstNumber = NaN;
+        txtResult.setText(null);
+        edtInput.setText(null);
+    }
 
-        btnNumber3.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "3");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    public void onDelClick(View view) {
+        String number = edtInput.getText().toString();
+        if (number.length() > 0) {
+            number = number.substring(0, number.length() - 1);
+        }
+        edtInput.setText(number);
+        edtInput.setSelection(edtInput.getText().length());
+    }
 
-        btnNumber4.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "4");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
```

Comment:
do not use SuppressLint without deep understanding why Lint highlight this method